medcat/cat.py

```diff
@@ -709,101 +708,139 @@ def train_supervised(self,
-                                     terminate_last=checkpoint.metadata["terminate_last"],
-                                     use_overlaps=checkpoint.metadata["use_overlaps"],
-                                     use_cui_doc_limit=checkpoint.metadata["use_cui_doc_limit"],
-                                     test_size=checkpoint.metadata["test_size"],
-                                     devalue_others=checkpoint.metadata["devalue_others"],
-                                     use_groups=checkpoint.metadata["use_groups"],
-                                     never_terminate=checkpoint.metadata["never_terminate"],
-                                     train_from_false_positives=checkpoint.metadata["train_from_false_positives"],
-                                     extra_cui_filter=checkpoint.metadata["extra_cui_filter"],
-                                     checkpoint=checkpoint,
-                                     resume_from_checkpoint=True)
+        if test_size == 0:
+            self.log.info("Running without a test set, or train==test")
+            test_set = data
+            train_set = data
+        else:
+            train_set, test_set, _, _ = make_mc_train_test(data, self.cdb, test_size=test_size)
+
+        if print_stats > 0:
+            fp, fn, tp, p, r, f1, cui_counts, examples = self._print_stats(test_set,
+                                                                           use_project_filters=use_filters,
+                                                                           use_cui_doc_limit=use_cui_doc_limit,
+                                                                           use_overlaps=use_overlaps,
+                                                                           use_groups=use_groups,
+                                                                           extra_cui_filter=extra_cui_filter)
+        if reset_cui_count:
+            # Get all CUIs
+            cuis = []
+            for project in train_set['projects']:
+                for doc in project['documents']:
+                    doc_annotations = self._get_doc_annotations(doc)
+                    for ann in doc_annotations:
+                        cuis.append(ann['cui'])
+            for cui in set(cuis):
+                if cui in self.cdb.cui2count_train:
+                    self.cdb.cui2count_train[cui] = 10
+
+        # Remove entities that were terminated
+        if not never_terminate:
+            for project in train_set['projects']:
+                for doc in project['documents']:
+                    doc_annotations = self._get_doc_annotations(doc)
+                    for ann in doc_annotations:
+                        if ann.get('killed', False):
+                            self.unlink_concept_name(ann['cui'], ann['value'])
+
+        latest_trained_step = checkpoint.count
+        current_epoch, current_project, current_document = self._get_training_start(train_set, latest_trained_step)
```

Comment:
We are duplicating code here and whatever change is made in the training iterator below has to be duplicated in this `_get_training_start`